textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -1,167 +1,222 @@
+     *           through all the fields. For each field, loop through all the
+     *           matches.
+     */
+    @Override
+    public ITuple getNextTuple() throws Exception {
+        if (attributeIndex < searchInAttributes.size()) {
+            IField dataField = dataTuple.getField(searchInAttributes.get(attributeIndex).getFieldName());
+
+            if (dataField instanceof StringField || dataField instanceof TextField) {
+                String fieldValue = (String) dataField.getValue();
+
+                matcher = pattern.matcher(fieldValue.toLowerCase());
+                // Get position of dict value in the field.
+                if (matcher.find(positionIndex) != false) {
+
+                    isPresent = true;
+                    spanIndexValue = matcher.start();
+                    // Increment positionIndex so that next search occurs from
+                    // new positionIndex.
+                    positionIndex = spanIndexValue + dictionaryValue.length();
+
+                    // Attribute attribute =
+                    // schema.getAttributes().get(attributeIndex);
+                    spanFieldName = searchInAttributes.get(attributeIndex).getFieldName(); // attribute.getFieldName();
+
+                    addSpanToSpanList(spanFieldName, spanIndexValue, positionIndex, dictionaryValue, dictionaryValue);
+                    return getNextTuple();
+
+                } else {
+                    // Increment the fieldIndex and call getNextTuple to search
+                    // in next field
+                    attributeIndex++;
+                    positionIndex = 0;
+                    return getNextTuple();
+                }
+            } else {
+                // If fieldType is not String. Presently only supporting string
+                // type in dictionary
+                attributeIndex++;
+                positionIndex = 0;
+                return getNextTuple();
+            }
+
+        } else if (attributeIndex == searchInAttributes.size() && isPresent) {
+            isPresent = false;
+            positionIndex = 0;
+            return getSpanTuple();
+        }
+        // Get the next document
+        else if ((dataTuple = operator.getNextTuple()) != null) {
+            attributeIndex = 0;
+            positionIndex = 0;
+            spanList.clear();
+
+            fields = dataTuple.getFields();
+            schema = dataTuple.getSchema();
+            spanSchema = createSpanSchema();
+            return getNextTuple();
+        }
+        // Get the next dictionary value
+        else if ((dictionaryValue = dictionary.getNextValue()) != null) {
+            // At this point all the documents in the dataStore are scanned
+            // and we need to scan them again for a different dictionary value
+            attributeIndex = 0;
+            positionIndex = 0;
+
+            dictionaryValueDup = dictionaryValue;

```

Comment:
I'm using dictionaryValueDup in order to lower case. This is because while returning the span tuple, I'm returning the original dictionary value, irrespective of it's case